### ISSUE TITLE
Evaluation on cityscapes (photo -> label) : what is the purpose of FCN-8s?

### ISSUE THREAD

**Issue by Ianlmgoddard** [junyanz/pytorch-CycleGAN-and-pix2pix#535]
Hello, we have successfully trained the pix2pix model you provide in the repo for photo->labels but now we are wondering how you evaluate. We've read your notes on evaluation but are slightly confused as to why we need the pre-trained FCN-8s model if we are trying to evaluate a pix2pix model that we have trained... 

Basically what is the purpose of this FCN-8s model?  Any help would be really appreciated and thanks for providing this code :)

**Comment by junyanz** [comment]
You don't need an FCN-8s for evaluating photo->labels direction. You can just compare it against ground truth human annotation. FCN-8s is used for evaluating labels-> photo direction.

**Comment by Ianlmgoddard** [comment]
Thanks for the quick response! 

I think I must be a bit confused about how to evaluate photo->labels then.  Where are the labelled (ground truth and predicted) tensors located? Do you advise that we use the scripts provided in eval_cityscapes?  

Sorry if Im asking something stupid, it's just that we were originally going to compare generated .pngs with the ground truth .pngs (like you state above) found in the cityscapes dataset which you provide code to download (again thank you for this) but we couldn't find a way to turn the .png files into label masks because there seems to be at least 240 different colors in the .png files. Surely there should be only 30 colours in the .pngs if there are only 30 possible classes in the cityscapes dataset? or maybe we're missing something...

Thanks again

**Comment by ZhangCZhen** [comment]
> how to evaluate photo->labels then

Hello, I have the same question about the evaluation of image2label generated results. How do you ultimately evaluate the results of semantic segmentation?
Looking forward to your answer, thanks a lot!

**Comment by junyanz** [comment]
We used to assign each RGB pixel to its closest color, and then use the segmentation evaluation code. Maybe you can also just predict class one-hot vector map instead rather than 3-channel RGB images.

**Comment by Kravrolens** [comment]
Sorry for disturbing you, because of the invalid link, could you please share "fcn-8s-cityscapes" weight to me? thank you!
